### ISSUE TITLE
BLOM fork with name BLOMGPU

### ISSUE THREAD

**Issue by monsieuralok** [NorESMhub/BLOM#81]
@TomasTorsvik @matsbn  @MichaelSchulzMETNO  Please can you create a BLOM fork with name BLOMGPU in NorESMhub? We need it for development on GPUs?

Thanks a lot
Alok

**Comment by TomasTorsvik** [comment]
@monsieuralok @matsbn @MichaelSchulzMETNO 
Hi,
I don't think it is easy (or even possible?) to fork a repository in the same organization space. Options include 
- create a git submodule : https://git-scm.com/book/en/v2/Git-Tools-Submodules
- duplicate the repository: https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/duplicating-a-repository
- fork in a different organization/account: https://github.community/t/alternatives-to-forking-into-the-same-account/10200

I suggest to set up an ad-hoc organization for the BLOMGPU fork that can be managed separately from NorESMhub. This seems to be the easiest solution, and it should be easy to give write access without involving NorESMhub. The person who set it up will be the owner of the repo, but others can be added as admins or given write access. I can set this up if you don't want to administrate this yourself.

**Comment by MichaelSchulzMETNO** [comment]
Alok did you mean branch, maybe?

**Comment by TomasTorsvik** [comment]
I think the idea was to make a fork. It seems that GPU acceleration will be enabled trough OpenMP in the near future (apparently, OpenACC is on the way out), and this may not be compatible with the existing OpenMP directives in BLOM. However, at the moment the plan is to continue with OpenACC a while longer and make a switch later on.

**Comment by agu002** [comment]
> Alok did you mean branch, maybe?

@MichaelSchulzMETNO no, we discussed yesterday about fork. I would then fork in my account. Thanks

**Comment by agu002** [comment]
> I think the idea was to make a fork. It seems that GPU acceleration will be enabled trough OpenMP in the near future (apparently, OpenACC is on the way out), and this may not be compatible with the existing OpenMP directives in BLOM. However, at the moment the plan is to continue with OpenACC a while longer and make a switch later on.

@TomasTorsvik That was the Idea. I would then fork it in my account. Thanks a lot for clarification. I noticed after forking I can change the repository name. Let see it doesn't conflict with BLOM and BLOMGPU in my account.

**Comment by TomasTorsvik** [comment]
@monsieuralok I think it would be a good idea to make a separate organization for the BLOMGPU fork
https://docs.github.com/en/github/getting-started-with-github/types-of-github-accounts#organization-accounts

This will make it easier to give write access and admin roles to other users. If you make it in your personal account, I think it can be more challenging to manage the repo later on.

**Comment by TomasTorsvik** [comment]
@monsieuralok See also
https://developers.sap.com/tutorials/webide-github-creating-org.html

**Comment by MichaelSchulzMETNO** [comment]
Sorry, its maybe not my business, but I dont see why you need a fork and a new organisation for this...You could also fork into you private account and cooperate on that fork.   I guess there are many ways to happiness.

**Comment by TomasTorsvik** [comment]
> Sorry, its maybe not my business, but I dont see why you need a fork and a new organisation for this...You could also fork into you private account and cooperate on that fork. I guess there are many ways to happiness.

Many ways indeed :)
I'm not saying this is the only way, just what I think would be easiest. Hopefully we can have some advanced support from Sigma2 (or Lumi admin, if we port to that system), and it may speed up the process if everyone has direct access to the same fork. At lest, that's my thinking.

**Comment by TomasTorsvik** [comment]
@monsieuralok - This issue has been hanging around for a while. Is this still work that is planned for the near future, or can we close this issue?
